.travis.yml

```diff
@@ -1,6 +1,8 @@
 language: c
+before_install: if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew upgrade && brew install gmp; fi
+install: wget https://raw.githubusercontent.com/ocaml/ocaml-ci-scripts/master/.travis-opam.sh
 script: bash -ex .travis-opam.sh
-sudo: false
+sudo: required
```

Comment:
I'll try to address these in order

1. Yup, these builds will be slower, but they'll follow the `ocaml-ci-scripts` master, which should be updated and current
2. I haven't had too much of an issue with build availability. travis does limit the number of parallel builds that `sudo` builds can do. I'm currently working on a little bit of `bash` that I'm trying to add to `ocaml-ci-scripts` that removes the need for `sudo` if you use the travis `apt` extension
3. `cache` is now available on `sudo` builds